### PR TITLE
Network Simulator API

### PR DESCRIPTION
**Pull request by KonradBkd**
## Subject

Experimental NetSim API candidate.

## Description

Public API for user-implemented network simulator participants.

## Instructions for review / testing

Review Code, Docs, Demos.

## Developer checklist (address before review)

- [x] Changelog.md updated
- [x] Prepared update for depending repositories
- [x] Documentation updated (public API changes only)
- [x] API docstrings updated (public API changes only)
- [x] Rebase &rarr; commit history clean
- [ ] Squash and merge &rarr; proper PR title

### CODE REVIEW

**Review by VDanielEdwards — COMMENTED** [review]
Review Part 1 of N

**Comment by KonradBkd** [comment]
> Review Part 1 of N

Done

File: SilKit/include/silkit/capi/InterfaceIdentifiers.h [hunk 1]
```diff
@@ -1,277 +1,377 @@
-/* Copyright (c) 2022 Vector Informatik GmbH
```
[the remark this answers is not in the document]
Comment (KonradBkd) [reply SilKit/include/silkit/capi/InterfaceIdentifiers.h:1]: Thanks for pointing that out, my git config was wrong indeed. We could consider adding `* text=auto` to the .gitattributes.

File: SilKit/IntegrationTests/Hourglass/Test_HourglassNetSim.cpp [hunk 2]
```diff
@@ -0,0 +1,1284 @@
...
+               != cControllerConfig.clusterParams->gMaxWithoutClockCorrectionPassive
+        || cppControllerConfig.clusterParams.gNumberOfMiniSlots != cControllerConfig.clusterParams->gNumberOfMiniSlots
+        || cppControllerConfig.clusterParams.gNumberOfStaticSlots
+               != cControllerConfig.clusterParams->gNumberOfStaticSlots
+        || cppControllerConfig.clusterParams.gPayloadLengthStatic
+               != cControllerConfig.clusterParams->gPayloadLengthStatic
+        || cppControllerConfig.clusterParams.gSyncFrameIDCountMax
+               != cControllerConfig.clusterParams->gSyncFrameIDCountMax)
+    {
+        return false;
```
Comment (MariusBgm) [inline SilKit/IntegrationTests/Hourglass/Test_HourglassNetSim.cpp:388]: I checked the code coverage using gcov + lcov here, and the `return false` branches in the matchers are never taken, so I guess this implicityl tests the assignCToCxx and assignCxxToC code paths.

File: SilKit/IntegrationTests/Hourglass/Test_HourglassNetSim.cpp [hunk 3]
```diff
@@ -0,0 +1,1284 @@
...
+    return true;
+}
+
+// -----------------------------
+// Matcher for Can
+// -----------------------------
+
+bool CompareCanFrame(const SilKit::Services::Can::CanFrame& cppFrame, const SilKit_CanFrame* cFrame)
+{
+    using namespace SilKit::Services::Can;
```
[the remark this answers is not in the document]
Comment (MariusBgm) [reply SilKit/IntegrationTests/Hourglass/Test_HourglassNetSim.cpp:65]: Maybe something to put on the TODO list for cleanups, then.

**Review by MariusBgm — APPROVED** [review]

**Review by VDanielEdwards — APPROVED** [review]
👍 LGTM!

File: Demos/NetworkSimulator/src/Can/MySimulatedCanController.hpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "silkit/experimental/netsim/all.hpp"
+#include "../MySimulatedNetwork.hpp"
+
+using namespace SilKit;
```
[the remark this answers is not in the document]
Comment (VDanielEdwards) [reply Demos/NetworkSimulator/src/Can/MySimulatedCanController.hpp]: Since this code lives in the demo application this is OK.

**Pull request merged**